scipy/stats/_continuous_distns.py

```diff
@@ -3616,6 +3616,39 @@ def _cdf(self, x, mu):
     def _stats(self, mu):
         return mu, mu**3.0, 3*np.sqrt(mu), 15*mu
 
+    def fit(self, data, *args, **kwds):
+
+        floc = kwds.pop('floc', None)
+        fscale = kwds.pop('fscale', None)
+        f0_s = kwds.pop('f0', None)
+
+        data = np.asarray(data)
+        if not np.isfinite(data).all():
+            raise RuntimeError("The data contains non-finite values.")
```

Comment:
This is somewhat temporary or unfinished. In #11988 we added `_check_fit_input_parameters`, but its behavior doesn't work for when super still needs to be called. 

- it wipes everything out of `kwds`
- raises error if anything is in `args`

I think we can either change it to remove those behaviors, add an option that lets it know to preserve data for fit methods like this one where super may be called, or not use it.